src/main/scala/com/nec/vectorengine/VeProcessImpl.scala

```diff
@@ -345,17 +347,19 @@ final case class WrappingVeo private (val node: Int,
 
   def unload(lib: LibraryReference): Unit = {
     withVeoProc {
-      val npath = Paths.get(lib.path).normalize
-      loadedLibRecords.get(npath.toString) match {
-        case Some(lib) =>
-          logger.info(s"[${handle.address}] Unloading library from the VE process: ${npath}...")
-          val result = veo.veo_unload_library(handle, lib.value)
-          require(result == 0, s"Failed to unload library from the VE process, got ${result} (library at: ${npath})")
-          // Remove only after the veo_unload_library() was successful
-          loadedLibRecords.remove(npath.toString)
+      loadedLibRecords.synchronized {
```

Comment:
Do we need this synchronization now that `withVeoProc` has its own lock?

Reply:
The lock used by `withVeoProc` is meant to ensure that the process is not closed while a process method is executing - it allows multiple threads to check `open` state and use the process but only one thread can access the `open` when it is updated (i.e. during `process.close`).  This is separate from `loadedLibRecords.synchronized`, which is meant to enforce loading libraries one at a time.